src/Fetch.js

```diff
@@ -276,7 +278,7 @@ function __emscripten_fetch_cache_data(db, fetch, data, onsuccess, onerror) {
 }
 #endif // ~FETCH_SUPPORT_INDEXEDDB
 
-function __emscripten_fetch_xhr(fetch, onsuccess, onerror, onprogress) {
+function __emscripten_fetch_xhr(fetch, onsuccess, onerror, onprogress, onheadersreceived) {
```

Comment:
Instead of adding a `onheadersreceived` event callback, let's add the more generic and direct `onreadystatechange` callback. That way we match the web API 1:1 and are more flexible with less code (otherwise in the future someone might be pressed to add that `onreadystatechange` callback, and then we'd have two: `onreadystatechange` and `onheadersreceived`, which would overlap)

Reply:
Done.